src/language-js/printer-estree.js

```diff
@@ -5204,7 +5204,9 @@ function printBinaryishExpressions(
     }
 
     const shouldInline = shouldInlineLogicalExpression(node);
-    const lineBeforeOperator = node.operator === "|>";
+    const lineBeforeOperator =
+      node.operator === "|>" &&
+      !hasLeadingOwnLineComment(options.originalText, node.right, options);
```

Comment:
It seems you accidentally include commits from #5015.

Reply:
yep, you're right. Thanks for noting! 😊